### PR TITLE
Fix "One Of" expanding #360

### LINKED ISSUE #360: Can not use custom types in "One Of"
I'm describing map POI api. So I have api method /map_markers which returns an array of markers. Each marker can be a poi or a cluster. I use such api description

```
FORMAT: 1A

# Test

# Data Structures

## Poi

+ name (string, required)
+ address (string, required)
+ phone (number, optional)

## Location

+ lon (number, required)
+ lat (number, required)

## MapMarker

+ One Of
    + Properties
        + kind: poi (string, required, fixed)
        + poi (Poi, required)
        + location (Location, required)
    + Properties
        + kind: cluster (string, required, fixed)
        + size (number, required)
        + location (Location, required)

# Group api

## POIs [/map_markers]

### Get POIs list [GET]

+ Response 200 (application/json)
    + Attributes
        + result (array[MapMarker], required, fixed)
```

but if I use custom type in `One Of` section (not just a number or string) - json schema will be wrong:

```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "result": {
      "type": "array",
      "items": [
        {
          "type": "object",
          "properties": {},
          "oneOf": [
            {
              "properties": {
                "kind": {
                  "type": "string",
                  "enum": [
                    "poi"
                  ]
                },
                "poi": {
                  "type": "object",
                  "properties": {}
                },
                "location": {
                  "type": "object",
                  "properties": {}
                }
              }
            },
            {
              "properties": {
                "kind": {
                  "type": "string",
                  "enum": [
                    "cluster"
                  ]
                },
                "size": {
                  "type": "number"
                },
                "location": {
                  "type": "object",
                  "properties": {}
                }
              }
            }
          ],
          "additionalProperties": false
        }
      ]
    }
  },
  "required": [
    "result"
  ]
}
```

Json schema contains empty objects instead of custom types.

### PR DESCRIPTION
**Pull request by goganchic**
Fixes #360

### CODE REVIEW

**Comment by pksunkara** [comment]
👍 Can you add a line to the changelog so that I can merge this? Thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
   [#127](https://github.com/apiaryio/drafter/issues/127)
   [snowcrash#200](https://github.com/apiaryio/snowcrash/issues/200)
 
+* Fix ability to use custom types in "One Of" sections.
```
Comment (pksunkara) [inline CHANGELOG.md]: `Fixes rendering when One Of contains member sections with named types`

**Comment by pksunkara** [comment]
Also can you squash the commits please? Thanks.

**Comment by goganchic** [comment]
@pksunkara done

**Comment by pksunkara** [comment]
@goganchic Thanks for the contribution. We really appreciate it. 😄 

Are you interested in any other bugs? Perhaps #193?

**Pull request merged**